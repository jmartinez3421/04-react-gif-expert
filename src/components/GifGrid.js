import React from 'react';
import PropTypes from 'prop-types';

import useFetchGifs from '../hooks/useFetchGifs';
import EraseButton from './EraseButton';
import GifGridItem from './GifGridItem';

const GifGrid = ({category, setCategories, categories}) => {
    const {data:images, loading} = useFetchGifs(category);       

    return (
        <div id={category}>
            <h3>{category}</h3><span><EraseButton category={category} categories={categories} setCategories={setCategories}/></span>
            <hr/>

            {loading && <p>Cargando...</p>}
            <div className='card-grid'>                
                <ol>
                    {images.map( img => {
                        return <GifGridItem 
                            key={img.id} 
                            {...img}
                        />
                    })}
                </ol>
            </div>
        </div>
    )

}

GifGrid.proptypes = {
    category: PropTypes.string.isRequired,
    setCategories: PropTypes.func.isRequired,
    categories: PropTypes.array.isRequired
}

export default GifGrid;
