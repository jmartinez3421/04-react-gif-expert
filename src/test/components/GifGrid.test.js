import "@testing-library/jest-dom";
import { shallow } from 'enzyme';
import React from 'react';
import GifGrid from '../../components/GifGrid';
import useFetchGifs from "../../hooks/useFetchGifs";
jest.mock("../../hooks/useFetchGifs");

describe('Pruebas sobre el componente <GifGrid />', () => {
    const categories = [];
    const setCategories = () => {jest.fn()};
    const category = '';

    test('el componente debe renderizarse correctamente', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow(<GifGrid category={category} categories={categories} setCategories={setCategories} />);

        expect(wrapper).toMatchSnapshot();

    });

    test('debe de mostrar items cuando se cargan imagenes con nuestro useFetchGifs', () => {

        const gifs = [{
            id: 'adfdfas',
            url: 'https://google.es',
            title: 'Google'
        },
        {
            id: '1234',
            url: 'https://google.es',
            title: 'Google'
        }];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        })

        const wrapper = shallow(<GifGrid category={category} categories={categories} setCategories={setCategories} />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe( gifs.length );
    })
})