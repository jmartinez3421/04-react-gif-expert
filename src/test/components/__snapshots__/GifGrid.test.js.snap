// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre el componente <GifGrid /> debe de mostrar items cuando se cargan imagenes con nuestro useFetchGifs 1`] = `
<div
  id=""
>
  <h3 />
  <span>
    <EraseButton
      categories={Array []}
      category=""
      setCategories={[Function]}
    />
  </span>
  <hr />
  <div
    className="card-grid"
  >
    <ol>
      <GifGridItem
        id="adfdfas"
        key="adfdfas"
        title="Google"
        url="https://google.es"
      />
      <GifGridItem
        id="1234"
        key="1234"
        title="Google"
        url="https://google.es"
      />
    </ol>
  </div>
</div>
`;

exports[`Pruebas sobre el componente <GifGrid /> el componente debe renderizarse correctamente 1`] = `
<div
  id=""
>
  <h3 />
  <span>
    <EraseButton
      categories={Array []}
      category=""
      setCategories={[Function]}
    />
  </span>
  <hr />
  <p>
    Cargando...
  </p>
  <div
    className="card-grid"
  >
    <ol />
  </div>
</div>
`;
